dmscripts/insert_dos_framework_results.py

```diff
@@ -0,0 +1,117 @@
+FAIL = "Fail"
+PASS = "Pass"
+DISCRETIONARY = "Discretionary"
+
+
+def insert_result(client, supplier_id, result, user):
+    try:
+        client.set_framework_result(supplier_id, 'digital-outcomes-and-specialists', result, user)
+        return"  Result set OK: {}".format(supplier_id)
+    except HTTPError as e:
+        return"  Error inserting result for {} ({}): {}".format(supplier_id, result, str(e))
+
+
+def get_submitted_drafts(client, supplier_id):
+    services = client.find_draft_services(supplier_id, framework='digital-outcomes-and-specialists')
+    services = services["services"]
+    submitted_services = [service for service in services if service["status"] == "submitted"]
+    return submitted_services
+
+
+def check_service_essentials(draft):
+    for essential in SERVICE_ESSENTIALS_MUST_BE_TRUE:
+        if essential in draft and draft[essential] is not True:
+            return FAIL
+    return PASS
+
+
+def check_declaration_answers(declaration_content, declaration):
+    if declaration['status'] != 'complete':
+        return FAIL
+
+    result = PASS
+    for (field_name) in declaration:
+        question_number = declaration_content.get_question(field_name).number \
+            if declaration_content.get_question(field_name) else -1
+        if (question_number in CORRECT_DECLARATION_RESPONSE_MUST_BE_TRUE
+           and declaration[field_name] is not True):
+            print(" Question {} must be True but is {}".format(question_number, declaration[field_name]))
+            result = FAIL
+
+        if (question_number in CORRECT_DECLARATION_RESPONSE_MUST_BE_FALSE
+           and declaration[field_name] is not False):
+            print(" Question {} must be False but is {}".format(question_number, declaration[field_name]))
+            result = FAIL
+
+        if (question_number in CORRECT_DECLARATION_RESPONSE_SHOULD_BE_FALSE
+           and declaration[field_name] is not False):
+            print(" Question {} should be False but is {}".format(question_number, declaration[field_name]))
+            if result == PASS:
+                result = DISCRETIONARY
+
+        if (question_number in CORRECT_DECLARATION_RESPONSES
+           and declaration[field_name] not in CORRECT_DECLARATION_RESPONSES[question_number]):
+            print(" Question {} has the wrong answer: {}".format(question_number, declaration[field_name]))
+            result = FAIL
+
+    return result
+
+
+def process_submitted_drafts(client, supplier_id, user):
+    submitted_drafts = get_submitted_drafts(client, supplier_id)
+    supplier_has_submitted_services = False
+
+    for draft in submitted_drafts:
+        draft_result = check_service_essentials(draft)
+        if draft_result == PASS:
+            supplier_has_submitted_services = True
+        else:
+            # Update the draft to be 'failed'
+            client.update_draft_service(draft['id'], {"status": "failed"}, user)
+
+    return supplier_has_submitted_services
```

Comment:
Discussed with Cath in real life - the behaviour here is correct and suppliers with no valid lots should be Failed.  The second script which generates the numbers for CCS then needs to show counts for `Draft`, `Complete-OK` and `Complete-Failed` services, which it can do by checking for the new `failed` status on drafts.